src/Features/Navigation/Favorite.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * WooCommerce Navigation Favorite
+ *
+ * @package Woocommerce Navigation
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\Navigation;
+
+use Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Contains logic for the WooCommerce Navigation menu.
+ */
+class Favorite {
+	/**
+	 * Class instance.
+	 *
+	 * @var Menu instance
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Array index of menu capability.
+	 *
+	 * @var int
+	 */
+	const META_NAME = 'navigation_favorites';
+
+	/**
+	 * Get class instance.
+	 */
+	final public static function instance() {
+		if ( ! static::$instance ) {
+			static::$instance = new static();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Init.
+	 */
+	public function init() {}
+
+	/**
+	 * Set given favorites string to the user meta data.
+	 *
+	 * @param string $current_user_id Current user id.
+	 * @param array  $favorites Array of favorite values to set.
+	 */
+	private static function set_meta_value( $current_user_id, $favorites ) {
+		Loader::update_user_data_field( $current_user_id, self::META_NAME, wp_json_encode( (array) $favorites ) );
+	}
+
+	/**
+	 * Add plugin to favorites
+	 *
+	 * @param string $plugin Identifier of plugin to add.
+	 */
+	public static function add_plugin( $plugin ) {
```

Comment:
I think `add_item()` or `add_category()` are slightly more semantic here.  I'm afraid there could be confusion that a plugin slug could be used in place here.

Reply:
Sounds good to me.